benchmark-base/src/main/java/org/conscrypt/EngineHandshakeBenchmark.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.conscrypt;
+
+import static org.conscrypt.TestUtils.doEngineHandshake;
+
+import java.nio.ByteBuffer;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLException;
+
+/**
+ * Benchmark comparing handshake performance of various engine implementations to conscrypt.
+ */
+public final class EngineHandshakeBenchmark {
+    /**
+     * Provider for the benchmark configuration
+     */
+    interface Config {
+        BufferType bufferType();
+        EngineType engineType();
+        String cipher();
+    }
+
+    private final EngineType engineType;
+    private final String cipher;
+
+    private final ByteBuffer clientApplicationBuffer;
+    private final ByteBuffer clientPacketBuffer;
+    private final ByteBuffer serverApplicationBuffer;
+    private final ByteBuffer serverPacketBuffer;
+
+    EngineHandshakeBenchmark(Config config) throws Exception {
+        engineType = config.engineType();
+        cipher = config.cipher();
+        final BufferType bufferType = config.bufferType();
```

Comment:
I don't think there's any reason to make this final, and we don't tend to make local variables final unless they have to be.